### PR TITLE
`WIP/DEPLAYED`: feat/KEEP-1241-bun-upgrade

### PR DESCRIPTION
**Pull request by suisuss**
- Replace Node.js with Bun for all Docker stages
- Update package.json scripts to use bun instead of tsx/pnpm
- Update Helm values for staging/prod (bun commands, remove NODE_OPTIONS)
- Update K8s manifests to use pre-built Bun-based images
- Add setup-bun to PR checks workflow
- Add bun.lock for dependency tracking

Bun's AsyncLocalStorage implementation supports Sentry v10+ tracing.

### CODE REVIEW

File: tests/e2e/graceful-shutdown.test.ts [hunk 1]
```diff
@@ -272,20 +272,27 @@ describe.skipIf(shouldSkip)("Graceful Shutdown E2E", () => {
       // Wait for graceful shutdown
       const result = await waitForExit(testProcess, 10_000);
 
-      // Exit code 1 = system termination (SIGTERM)
-      expect(result.exitCode).toBe(1);
+      // Exit code 1 (Node.js) or 143 (Bun: 128 + SIGTERM) = system termination
+      expect([1, 143]).toContain(result.exitCode);
```
Comment (joelorzet) [inline tests/e2e/graceful-shutdown.test.ts:276]: Shouldn’t we just rely on Bun’s exit code? Since we’re completely migrating the runner, I think this could potentially add edge cases.

**Review by joelorzet — COMMENTED** [review]
Honestly, it looks amazing. I’d definitely like to give it a try. That said, I’d prefer to do it in a separate environment to keep staging safe for a few days, since this might introduce unexpected issues as we continuously fetch updates from Vercel’s repo.

File: Dockerfile [hunk 2]
```diff
@@ -1,41 +1,51 @@
...
+# Strategy:
+# - Node.js for production app (full Sentry tracing support)
+# - Bun for background services (faster startup, native TypeScript)
+#
+# See docs/keeperhub/KEEP-1241/ for migration notes and known issues.
+#
+# =============================================================================
+# Stage 1: Dependencies (Bun for fast installs)
+# =============================================================================
+FROM oven/bun:1.2-alpine AS deps
```
Comment (OleksandrUA) [inline Dockerfile:12]: @suisuss I see `1.3` is available. Should we give newest version a try?

**Review by OleksandrUA — APPROVED** [review]

**Comment by eskp** [comment]
Postponed
